src/java/org/apache/cassandra/index/sai/disk/v1/V1OnDiskFormat.java

```diff
@@ -0,0 +1,256 @@
+                                       indexDescriptor,
+                                       index.getIndexContext(),
+                                       rowMapping);
+    }
+
+    @Override
+    public boolean validatePerSSTableComponents(IndexDescriptor indexDescriptor, boolean checksum)
+    {
+        for (IndexComponent indexComponent : perSSTableComponents())
+        {
+            if (isNotBuildCompletionMarker(indexComponent))
+            {
+                try (IndexInput input = indexDescriptor.openPerSSTableInput(indexComponent))
+                {
+                    if (checksum)
+                        SAICodecUtils.validateChecksum(input);
+                    else
+                        SAICodecUtils.validate(input);
+                }
+                catch (Throwable e)
+                {
+                    if (logger.isDebugEnabled())
+                    {
+                        logger.debug(indexDescriptor.logMessage("{} failed for index component {} on SSTable {}"),
+                                     (checksum ? "Checksum validation" : "Validation"),
+                                     indexComponent,
+                                     indexDescriptor.descriptor);
+                    }
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean validatePerIndexComponents(IndexDescriptor indexDescriptor, IndexContext indexContext, boolean checksum)
+    {
+        for (IndexComponent indexComponent : perIndexComponents(indexContext))
+        {
+            if (isNotBuildCompletionMarker(indexComponent))
+            {
+                try (IndexInput input = indexDescriptor.openPerIndexInput(indexComponent, indexContext))
+                {
+                    if (checksum)
+                        SAICodecUtils.validateChecksum(input);
+                    else
+                        SAICodecUtils.validate(input);
+                }
+                catch (Throwable e)
+                {
+                    if (logger.isDebugEnabled())
+                    {
+                        logger.debug(indexDescriptor.logMessage("{} failed for index component {} on SSTable {}"),
+                                     (checksum ? "Checksum validation" : "Validation"),
+                                     indexComponent,
+                                     indexDescriptor.descriptor);
+                    }
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public Set<IndexComponent> perSSTableComponents()
+    {
+        return PER_SSTABLE_COMPONENTS;
+    }
+
+    @Override
+    public Set<IndexComponent> perIndexComponents(IndexContext indexContext)
+    {
+        return LITERAL_COMPONENTS;
+    }
+
+    @Override
+    public int openFilesPerSSTable()
+    {
+        return 4;
```

Comment:
nit: If we leave this hard-coded, let's at least provide a comment, similar to the one in `openFilesPerIndex()`.